lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -538,6 +546,80 @@ ArrayAttr TypeLoweringVisitor::filterAnnotations(MLIRContext *ctxt,
   return ArrayAttr::get(ctxt, retval);
 }
 
+LogicalResult TypeLoweringVisitor::partitionSymbols(
+    hw::InnerSymAttr sym, ArrayRef<FlatBundleFieldEntry> fields,
+    SmallVectorImpl<hw::InnerSymAttr> &newSyms, Location errorLoc) {
+
+  // No symbol, nothing to partition.
+  if (!sym)
+    return success();
+
+  auto *context = sym.getContext();
+
+  // Helper for getting max fieldID from const FIRRTLBaseType's.
+  auto getMaxFieldID = [](const FIRRTLBaseType type) {
+    return FIRRTLBaseType(type).getMaxFieldID();
```

Comment:
Is this just working around const?

Reply:
Yep.  For whatever reason our versions aren't const but the interface versions are.